common/src/main/java/com/wynntils/commands/LootrunCommand.java

```diff
@@ -0,0 +1,430 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.commands;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.arguments.StringArgumentType;
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.suggestion.SuggestionProvider;
+import com.mojang.brigadier.tree.LiteralCommandNode;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.mc.utils.commands.CommandBase;
+import com.wynntils.wc.utils.lootrun.LootrunUtils;
+import com.wynntils.wc.utils.lootrun.objects.LootrunUncompiled;
+import com.wynntils.wc.utils.lootrun.objects.RecordingInformation;
+import it.unimi.dsi.fastutil.Pair;
+import it.unimi.dsi.fastutil.objects.ObjectObjectImmutablePair;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Stream;
+import net.minecraft.ChatFormatting;
+import net.minecraft.Util;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.commands.Commands;
+import net.minecraft.commands.SharedSuggestionProvider;
+import net.minecraft.commands.arguments.ComponentArgument;
+import net.minecraft.commands.arguments.coordinates.BlockPosArgument;
+import net.minecraft.core.BlockPos;
+import net.minecraft.network.chat.*;
+import net.minecraft.world.phys.Vec3;
+
+public class LootrunCommand extends CommandBase {
```

Comment:
it doesn't seem logical to me that so much of the lootrun logic is handled here, especially the way variables in `LootrunUtils` are being directly assigned. if we have a utils class, i feel like we should be doing all that logic inside of it via function calls instead of this

Reply:
Refactored this code, hope its better now